src/components/field/index.js

```diff
@@ -140,7 +140,7 @@ export default class TextField extends PureComponent {
       newState.error = error
     }
 
-    if (value !== state.text) {
+    if (value && value !== state.text) {
```

Comment:
Here's what was happening:

- We made the field uncontrolled (i.e. not setting `value`)
- This gets called with `value: undefined`
- It's different from `state.text` which is '12345' or whatever
- `undefined` overwrites '12345'
- The animation is sad

Reply:
Although, philosophically, perhaps I should allow a value of `null` but not `undefined`? Is that the different between not-set and set-to-null?

Reply:
First off, I just want to make sure this works if "0" is provided as a value. Onto this null vs. undefined discussion

If you allow a value of null and not undefined, that means the value must always be intentionally set to null. Undefined is the default value for props that aren't declared. Here's my flow suggestion on `FormTextInput` for making sure all cases are handled:

```
type FormTextInputCommonProps = {
  style?: ViewStyleProp,
  error?: string | React.Node,
  label?: string | React.Node,
};

type ValueOrMask = { value: ?(string | number) } | { mask: string };

export type Props = React.ElementConfig<typeof TextField> &
  FormTextInputCommonProps &
  ValueOrMask;
```

The `value: ?(string | number)` there means that the value prop must be passed down. The leading `?` means that the prop's value can be null or undefined in addition to some defined value. So in this case, the `value` prop must be a `string`, a `number`, `null`, or `undefined`. It can't be left out

Also, the Props are being exported for use in flavors of `FormTextInput`, like `FormInertTextInput`